### PR TITLE
S3_formValidateService

### PR DESCRIPTION
**Pull request by Vannou28**
class made to control and validate case for form view

### CODE REVIEW

**Review by khadimb — DISMISSED** [review]
nickel

File: README class FormValidation.txt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        //affichage erreur : array
+        var_dump($validation->getErrors());
+
+        // affichage erreurs formater html : string
+        if ($validation->isNotErrors() === false) {
+            echo $validation->displayErrorsHtml();
+        }
+        
+    }
+}
```
Comment (LucyMerlier) [inline README class FormValidation.txt:50]: what's this file doing here?

File: src/Services/FormValidation.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\Services;
```
Comment (LucyMerlier) [inline src/Services/FormValidation.php:3]: good practive is to use singular ("App\Service" and "src/Service")

File: src/Services/FormValidation.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    public function sentenceEmpty(string $sentences, string $messageError): bool
+    {
+        if (empty($sentences)) {
+            $this->setErrors($messageError);
+            return true;
+        } else {
+            return false;
+        }
+    }
```
Comment (LucyMerlier) [inline src/Services/FormValidation.php:17]: why does this method return a bool when all the others return void?

File: src/Services/FormValidation.php [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    {
+        if (strlen($word) < $length) {
+            $this->setErrors($messageError);
+        }
+    }
+
+    private function setErrors(string $errors): void
+    {
+        $this->errors[] = $errors;
+    }
```
Comment (LucyMerlier) [inline src/Services/FormValidation.php:57]: this looks like it doesn't set errors, it looks like it adds an error ;)

File: src/Services/FormValidation.php [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        return $this->errors;
+    }
+
+    public function isNotErrors(): bool
+    {
+        if (empty($this->errors)) {
+            return true;
+        }
+        return false;
+    }
```
Comment (LucyMerlier) [inline src/Services/FormValidation.php:70]: ?

File: src/Services/FormValidation.php [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+    public function displayErrorsHtml(): string
+    {
+        $html = '<ul>';
+        foreach ($this->getErrors() as $error) {
+            $html .= '<li>' . $error . '</li>';
+        }
+        $html .= '</ul>';
+        return $html;
+    }
```
Comment (LucyMerlier) [inline src/Services/FormValidation.php:80]: you should not handle html tags outside of twig templates

File: src/Services/FormValidation.php [hunk 7]
```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\Services;
+
+class FormValidation
+{
+    protected array $errors = [];
```
Comment (LucyMerlier) [inline src/Services/FormValidation.php:7]: why is it protected? your class does not have any child class?

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments

File: README class FormValidation.txt (hunk 1 above)
Comment (Vannou28) [reply README class FormValidation.txt:50]: c'est pour les copians

File: src/Services/FormValidation.php (hunk 2 above)
Comment (Vannou28) [reply src/Services/FormValidation.php:3]: si on fait plusieurs services !!!

File: src/Services/FormValidation.php (hunk 3 above)
Comment (Vannou28) [reply src/Services/FormValidation.php:17]: car j'en est besoin en bool

File: src/Services/FormValidation.php (hunk 4 above)
Comment (Vannou28) [reply src/Services/FormValidation.php:57]: on enregistre une erreur dans un tableau donc on fait un set

File: src/Services/FormValidation.php (hunk 5 above)
Comment (Vannou28) [reply src/Services/FormValidation.php:70]: ca peut servir ;-)))

File: src/Services/FormValidation.php (hunk 6 above)
Comment (Vannou28) [reply src/Services/FormValidation.php:80]: je trouvais ça pratique pour les copains

File: src/Services/FormValidation.php (hunk 7 above)
Comment (Vannou28) [reply src/Services/FormValidation.php:7]: ok je vais le mettre le private

**Review by Erdnaxela45 — APPROVED** [review]

**Review by LucyMerlier — APPROVED** [review]

**Pull request merged**